test/AdventureGoldGovernance.t.sol

```diff
@@ -0,0 +1,187 @@
+    error OnlyTransfersToFromContract();
+
+    /// @notice Error emitted when the transfer function is called
+    error NonTransferable();
+
+    /**
+     * @dev Indicates an error related to the current `balance` of a `sender`. Used in transfers.
+     * @param sender Address whose tokens are being transferred.
+     * @param balance Current balance for the interacting account.
+     * @param needed Minimum amount required to perform a transfer.
+     */
+    error ERC20InsufficientBalance(address sender, uint256 balance, uint256 needed);
+
+    function setUp() public {
+        mainnetFork = vm.createFork(vm.envString("ETHEREUM_MAINNET_RPC_URL"));
+        vm.selectFork(mainnetFork);
+        // Roll to a block number post AdventureGoldGovernance deployment
+        vm.rollFork(19834161);
+        adventureGoldGovernance = new AdventureGoldGovernance();
+        adventureGold = IERC20(0x32353A6C91143bfd6C7d363B546e62a9A2489A20);
+    }
+
+    function test_AdventureGoldAddress() public {
+        assertEq(adventureGoldGovernance.AGLD_TOKEN_ADDRESS(), 0x32353A6C91143bfd6C7d363B546e62a9A2489A20);
+    }
+
+    function test_DepositAndWithdraw() public {
+        // Starting balance of AGLD
+        uint256 startingBalance = adventureGold.balanceOf(testingAddress);
+        // Starting balance of Adventure Gold Governance
+        assertEq(adventureGoldGovernance.balanceOf(testingAddress), 0);
+        assertEq(adventureGoldGovernance.totalSupply(), 0);
+
+        // 100 tokens (which uses 18 decimals)
+        uint256 amount = 100 * 10 ** 18;
+
+        // Deposit 100 AGLD tokens
+        vm.startPrank(testingAddress);
+        adventureGold.approve(address(adventureGoldGovernance), amount);
+        adventureGoldGovernance.deposit(amount);
+
+        // Check balances after deposit
+        assertEq(adventureGold.balanceOf(testingAddress), startingBalance - amount);
+        assertEq(adventureGoldGovernance.balanceOf(testingAddress), amount);
+        assertEq(adventureGoldGovernance.totalSupply(), amount);
+
+        // Withdraw 100 AGLD tokens
+        adventureGoldGovernance.withdraw(amount);
+
+        // Check balances after withdrawal
+        assertEq(adventureGold.balanceOf(testingAddress), startingBalance);
+        assertEq(adventureGoldGovernance.balanceOf(testingAddress), 0);
+        assertEq(adventureGoldGovernance.totalSupply(), 0);
+
+        vm.stopPrank();
+    }
+
+    function test_CannotTransferExceptToOrFromContract() public {
+        // 100 tokens (which uses 18 decimals)
+        uint256 amount = 100 * 10 ** 18;
+
+        // Deposit 100 AGLD tokens
+        vm.startPrank(testingAddress);
+        adventureGold.approve(address(adventureGoldGovernance), amount);
+        adventureGoldGovernance.deposit(amount);
+
+        // Give approvals for transferFrom
+        adventureGoldGovernance.approve(testingAddress2, amount);
+        adventureGoldGovernance.approve(address(adventureGoldGovernance), amount);
```

Comment:
Related comment: https://github.com/WillPapper/Adventure-Gold-Governance-Wrapper/pull/2/files#r1624607027

Do we even need to allow approvals in the first place?